src/protocols/wallet-provider/maskbook.ts

```diff
@@ -0,0 +1,68 @@
+import Web3 from 'web3'
+import type { HttpProvider } from 'web3-core'
+import type { HttpProviderOptions } from 'web3-core-helpers'
+import type { EthereumNetwork } from '../../plugins/Wallet/database/types'
+import { getNetworkSettings } from '../../plugins/Wallet/UI/Developer/EthereumNetworkSettings'
+import { getManagedWallets } from '../../plugins/Wallet/wallet'
+import { web3 } from '../../plugins/Wallet/web3'
+import { currentLocalWalletEthereumNetworkSettings } from '../../settings/settings'
+import type { WalletProvider } from './index'
+
+type SwitchableProvider = { switch(network: EthereumNetwork): void; disconnect(): void }
+function SwitchableProvider(opts?: HttpProviderOptions): SwitchableProvider & HttpProvider {
```

Comment:
I think you can achieve the goal with a local variable `pool`. Class and proxy make it harder to understand. Connecting and disconnecting can be easily done in `MaskbookProvider`.

Reply:
I'm not using class, what's the problem here?